src/visit.rs

```diff
@@ -139,7 +139,7 @@ pub fn visit<'a, Err, V: Visitor<'a, Err>>(v: &mut V, c: CborValue<'a>) -> Resul
             let mut iter = Iter::new(bytes, length.map(|x| x * 2));
             let mut is_first = true;
             while let Some(key) = iter.next() {
-                if let Some(Str(key)) = tagged_value(key.as_slice()).map(|x| x.kind) {
+                if let Some(key) = tagged_value(key.as_slice()) {
```

Comment:
As discussed: this can stay as is, but we should advertise that indexing is only supported for strings, byte strings, and 64bit numbers — quite possibly underlined by adding a high-level API based on a type class that enforces this constraint. Byte strings can contain valid CBOR, but our indexing logic won’t care, it will only compare the bytes.

Reply:
I added a comment on the visitor about how to visit complex keys.